presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java

```diff
@@ -124,6 +143,38 @@ public static void mergeVarianceState(VarianceState state, VarianceState otherSt
         state.setMean(newMean);
     }
 
+    public static void mergeCentralMomentsState(CentralMomentsState state, CentralMomentsState otherState)
+    {
+        long na = state.getCount();
+        long nb = otherState.getCount();
+
+        checkArgument(nb >= 0, "count is negative");
+        if (nb == 0) {
+            return;
+        }
+
+        double m1a = state.getM1();
+        double m2a = state.getM2();
+        double m3a = state.getM3();
+        double m1b = otherState.getM1();
+        double m2b = otherState.getM2();
+        double m3b = otherState.getM3();
+        long n = na + nb;
```

Comment:
`double n = na + nb;`

At the scale we are looking at, using `long` as the type will likely overflow with `n*n` and will definitely cause problem with `n*n*n`.